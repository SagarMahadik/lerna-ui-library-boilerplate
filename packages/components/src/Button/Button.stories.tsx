import React from "react";
import { Button, ButtonProps } from "./Button";
import { ComponentStory, ComponentMeta } from "@storybook/react";

export default {
	title: "Components/Button",
	component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args: ButtonProps) => (
	<Button {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
	label: "Primary Button",
	onClick: () => {
		console.log("Primary button clicked");
	},
};

export const Secondary = Template.bind({});
Secondary.args = {
	label: "Secondary Button",
	onClick: () => {
		console.log("Secondary button clicked");
	},
};
