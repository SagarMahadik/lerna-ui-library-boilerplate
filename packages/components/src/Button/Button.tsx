import * as React from "react";
import { Button as StyledButton, ButtonText } from "@bennytestui1/styles";

interface ButtonProps {
	label: string;
	onClick: () => void;
}

const Button = ({ label, onClick }: ButtonProps) => {
	return (
		<StyledButton onClick={onClick}>
			<ButtonText>{label}</ButtonText>
		</StyledButton>
	);
};

export { Button };
export type { ButtonProps };
export default Button;
