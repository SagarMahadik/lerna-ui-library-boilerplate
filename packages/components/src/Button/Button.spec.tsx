/**
 * @jest-environment jsdom
 */

const { render } = require("@testing-library/react");
import { Button } from "./Button";
import React from "react";
// import "@testing-library/jest-dom";

// Mock the styled component
describe("Button", () => {
	it("should render the button", () => {
		const { container } = render(
			<Button onClick={() => {}} label="Click me" />
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
