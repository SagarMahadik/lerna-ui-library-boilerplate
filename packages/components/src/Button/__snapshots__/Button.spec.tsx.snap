// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the button 1`] = `
<div
  class="sc-bcXHqe jlwint"
>
  <h1
    class="sc-gswNZR dsEACx"
  >
    Click me
  </h1>
</div>
`;
